service/worker/archiver/client_worker.go

```diff
@@ -53,17 +53,20 @@ type (
 
 	// BootstrapContainer contains everything need for bootstrapping
 	BootstrapContainer struct {
-		PublicClient         workflowserviceclient.Interface
-		MetricsClient        metrics.Client
-		Logger               log.Logger
-		ClusterMetadata      cluster.Metadata
-		HistoryManager       persistence.HistoryManager
-		HistoryV2Manager     persistence.HistoryV2Manager
-		Blobstore            blobstore.Client
-		DomainCache          cache.DomainCache
-		Config               *Config
-		HistoryBlobReader    HistoryBlobReader // this is only set in testing code
-		HistorySizeEstimator SizeEstimator     // this is only set in testing code
+		PublicClient     workflowserviceclient.Interface
+		MetricsClient    metrics.Client
+		Logger           log.Logger
+		ClusterMetadata  cluster.Metadata
+		HistoryManager   persistence.HistoryManager
+		HistoryV2Manager persistence.HistoryV2Manager
+		Blobstore        blobstore.Client
+		DomainCache      cache.DomainCache
+		Config           *Config
+
+		// the following are only set in testing code
+		HistoryBlobReader     HistoryBlobReader
```

Comment:
if this is only used in testing, can you make these things private - assuming the tests that are using it are within the same package ?

Reply:
We cannot because they will then not get propagated as part of the context.